src/MosekTools.jl

```diff
@@ -384,77 +384,93 @@ function MOI.optimize!(m::Optimizer)
+                Float64[],                                     # slc
+                Float64[],                                     # suc
+                Float64[],                                     # y
             ),
         )
     end
     if Mosek.solutiondef(m.task, Mosek.MSK_SOL_BAS)
         push!(
             m.solutions,
             MosekSolution(
-                Mosek.MSK_SOL_BAS,
-                Mosek.getsolsta(m.task, Mosek.MSK_SOL_BAS),
-                Mosek.getprosta(m.task, Mosek.MSK_SOL_BAS),
-                Mosek.getskx(m.task, Mosek.MSK_SOL_BAS),
-                Mosek.getxx(m.task, Mosek.MSK_SOL_BAS),
+                Mosek.MSK_SOL_BAS,                              # whichsol
+                Mosek.getsolsta(m.task, Mosek.MSK_SOL_BAS),     # solsta
+                Mosek.getprosta(m.task, Mosek.MSK_SOL_BAS),     # prosta
+                Mosek.getskx(m.task, Mosek.MSK_SOL_BAS),        # xxstatus
+                Mosek.getxx(m.task, Mosek.MSK_SOL_BAS),         # xx
+                # MSK_SOL_BAS means a basic solution. It cannot have PSD
+                # matrices.
                 # See https://github.com/jump-dev/MosekTools.jl/issues/71
-                Float64[], #matrix_solution(m, Mosek.MSK_SOL_BAS),
-                Mosek.getslx(m.task, Mosek.MSK_SOL_BAS),
-                Mosek.getsux(m.task, Mosek.MSK_SOL_BAS),
-                Float64[],
-                Float64[],
-                Mosek.getskc(m.task, Mosek.MSK_SOL_BAS),
-                Mosek.getxc(m.task, Mosek.MSK_SOL_BAS),
-                Mosek.getslc(m.task, Mosek.MSK_SOL_BAS),
-                Mosek.getsuc(m.task, Mosek.MSK_SOL_BAS),
-                Mosek.gety(m.task, Mosek.MSK_SOL_BAS),
+                Float64[],                                      # barxk
+                Mosek.getslx(m.task, Mosek.MSK_SOL_BAS),        # slx
+                Mosek.getsux(m.task, Mosek.MSK_SOL_BAS),        # sux
+                Float64[],                                      # snx
+                Float64[],                                      # doty
+                Mosek.getskc(m.task, Mosek.MSK_SOL_BAS),        # cstatus
+                Mosek.getxc(m.task, Mosek.MSK_SOL_BAS),         # xc
+                Mosek.getslc(m.task, Mosek.MSK_SOL_BAS),        # slc
+                Mosek.getsuc(m.task, Mosek.MSK_SOL_BAS),        # suc
+                Mosek.gety(m.task, Mosek.MSK_SOL_BAS),          # y
             ),
         )
     end
-    # We need to sort the solutions, so that an optimal one is first (if it exists).
-    sort!(
-        m.solutions;
-        by = x -> x.solsta in
-        [Mosek.MSK_SOL_STA_OPTIMAL, Mosek.MSK_SOL_STA_INTEGER_OPTIMAL],
-        rev = true,
-    )
+    # We need to sort the solutions, so that an optimal one is first (if it
+    # exists). The priority is:
+    #  1. MSK_SOL_STA_INTEGER_OPTIMAL or MSK_SOL_STA_OPTIMAL
+    #  2. MSK_SOL_ITG, MSK_SOL_BAS, MSK_SOL_ITR
```

Comment:
This is the first element of the tuple so it should be the first in this enumerate no ?

Reply:
Ooops. Fixed